interop_binaries/src/testcontainer.rs

```diff
@@ -0,0 +1,89 @@
+//! Provides a testcontainer image to allow container-based testing of Janus.
+
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::{
+    io::{Read, Write},
+    process::{Command, Stdio},
+    sync::Mutex,
+    thread,
+};
+use testcontainers::{core::WaitFor, Image};
+
+// INTEROP_AGGREGATOR_IMAGE_BYTES / interop_aggregator.tar are created by this package's build.rs.
+const INTEROP_AGGREGATOR_IMAGE_BYTES: &[u8] =
+    include_bytes!(concat!(env!("OUT_DIR"), "/interop_aggregator.tar"));
```

Comment:
How big is the interop aggregator image? If we pull it in with `include_bytes!`, does it all get loaded into memory at once, or will the `child_stdin.write_all(INTEROP_AGGREGATOR_IMAGE_BYTES)` call below gracefully read a chunk of it at a time?

Reply:
Oh dear, the tar file from when I built it is 750MB! It would get mapped in as part of the `.rdata` section, I think. The two biggest contributors are the final layer, with `janus_interop_aggregator` only, and the layer that installed postgres, perl, etc.

Reply:
Streaming the bytes from a `.tar` on disk and then into `child_stdin` seems like it ought to be about the same as streaming the bytes from the binary on disk, but yeah, maybe 750 MB in `.rdata` is a bit much.

Reply:
(that said, no objection from me if we want to try to shrink these binaries/images somehow. ~330 MB for the amount of functionality in an aggregator, or ~750 MB for aggregator + DB, is wild to me.)

Reply:
I checked out 10c92e7, cleaned my target directory, and triggered the OOM killer a few times when trying to build, either with `cargo build` or `cargo test`. For example:

```
error: could not compile `interop_binaries`

Caused by:
  process didn't exit successfully: `rustc --crate-name interop_binaries --edition=2021 interop_binaries/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="testcontainer"' -C metadata=fd8a365b13b98fba -C extra-filename=-fd8a365b13b98fba --out-dir /home/david/Code/ppm/janus/target/debug/deps -C incremental=/home/david/Code/ppm/janus/target/debug/incremental -L dependency=/home/david/Code/ppm/janus/target/debug/deps --extern ... -L native=/home/david/Code/ppm/janus/target/debug/build/ring-bce3495b5618ba54/out` (signal: 9, SIGKILL: kill)
```

```
kernel: Out of memory: Killed process 287248 (rustc) total-vm:9795372kB, anon-rss:3633724kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:13796kB oom_score_adj:0
```

```
kernel: Out of memory: Killed process 284952 (rustc) total-vm:25262432kB, anon-rss:9123824kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:27788kB oom_score_adj:0
```

Building succeeded eventually after some retries. I figure the SIGKILL in the CI logs is due to the same cause. The `interop_aggregator.tar` file is 750M, as before. I figure rustc and LLVM are both making several copies of the byte array as it passes through the pipeline.

How about we add `RUSTFLAGS="-C debuginfo=0"` in the build.rs invocation, via a new argument?